src/Pgvector.EntityFrameworkCore/VectorFunctionTranslatorPlugin.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
+using Microsoft.EntityFrameworkCore.Storage;
+using System.Reflection;
+
+namespace Pgvector.EntityFrameworkCore;
+
+public class VectorFunctionTranslatorPlugin : IMethodCallTranslatorPlugin
+{
+    public VectorFunctionTranslatorPlugin(ISqlExpressionFactory sqlExpressionFactory, ITypeMappingSource typeMappingSource)
+    {
+        Translators = new[]
+        {
+            new VectorFunctionTranslator(sqlExpressionFactory, typeMappingSource),
+        };
+    }
+
+    public virtual IEnumerable<IMethodCallTranslator> Translators { get; }
+
+    private class VectorFunctionTranslator : IMethodCallTranslator
+    {
+        private readonly ISqlExpressionFactory _sqlExpressionFactory;
+        private readonly ITypeMappingSource _typeMappingSource;
+
+        private static readonly MethodInfo _methodEuclideanDistance = typeof(VectorExtensions)
+            .GetRuntimeMethod(nameof(VectorExtensions.EuclideanDistance), new[]
+            {
+                typeof(Vector),
+                typeof(Vector),
+            })
+            ?? throw new InvalidOperationException($"Method {nameof(VectorExtensions.EuclideanDistance)} is not found");
```

Comment:
The EF standard is to just use the bang (!) operator here rather than throwing. In the off chance that something changes and this fails for some reason, you don't want to completely block the application by throwing, when it might not even be using these in actual queries.